### PR TITLE
Parametrized tests

### PR DESCRIPTION
**Pull request by Kosmoastronauta**

### CODE REVIEW

File: src/main/java/Traffic/Reservation.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public int reservSpace(int number_of_space, Time newTime)
 
     public boolean isFree(int number_of_space)
```
Comment (bkita) [inline src/main/java/Traffic/Reservation.java]: Is it for parking space or entire parking? Can You please rename method to more descriptive?

File: src/main/java/Traffic/Reservation.java [hunk 2]
```diff
@@ -84,12 +84,16 @@ else if(!this.times.containsKey(number_of_space))
             return false;
     }
 
+
+
     public void reset()
```
Comment (bkita) [inline src/main/java/Traffic/Reservation.java]: same as above

File: src/main/java/Traffic/Reservation.java [hunk 3]
```diff
@@ -84,12 +84,16 @@ else if(!this.times.containsKey(number_of_space))
...
     {
         for(int i=0; i<this.parking.number_of_spaces; i++)
         {
             this.spaces[i] = false;
+
```
Comment (bkita) [inline src/main/java/Traffic/Reservation.java]: delete empty line

File: src/main/java/Traffic/Reservation.java [hunk 4]
```diff
@@ -103,7 +107,7 @@ public boolean empty()
 
     protected boolean possibleToReserve(Time oldTime, Time newTime)
```
Comment (bkita) [inline src/main/java/Traffic/Reservation.java]: if method returns boolean, it should start from is -> isPossibleToReserve

Comment (bkita) [reply src/main/java/Traffic/Reservation.java]: maybe isParkingPlacePossibleToReserve

File: src/main/java/Traffic/Time.java [hunk 5]
```diff
@@ -37,6 +37,34 @@ public int valueTo()
...
+        if(validationHour(hour) && validationMin(min))
+        {
+            return hour * 60 + min;
+        }
+
+        else
+            throw new InvalidParameterException("Inproper input");
+    }
+
+    protected boolean validation(int hourFrom, int minFrom, int hourTo, int minTo)
```
Comment (bkita) [inline src/main/java/Traffic/Time.java]: validation of what?

File: src/main/java/Traffic/Time.java [hunk 6]
```diff
@@ -37,6 +37,34 @@ public int valueTo()
...
+            {
+                return true;
+            }
+            else
+                return false;
+        }
+        else
+            return false;
+    }
+
```
Comment (bkita) [inline src/main/java/Traffic/Time.java:65]: you dont need that many empty spaces

File: src/test/java/Traffic/ParametrizedRezerwationTest.java [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+    }
+
+    @After
+    public void clear() {
+        reservation.reset();
+    }
+
+
+    public ParametrizedRezerwationTest(int hourFrom, int minFrom, int hourTo, int minTo)
```
Comment (bkita) [inline src/test/java/Traffic/ParametrizedRezerwationTest.java:31]: Reservation ?

File: src/test/java/Traffic/ParametrizedRezerwationTest.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        //When
+        reservation.reservSpace(1, new Time(currentHourFrom,currentMinFrom,currentHourTo,currentMinTo));
+
+        //Then
+        Assert.assertEquals(reservation.avaliable(2,new Time(currentHourFrom,currentMinFrom,currentHourTo,currentMinTo)),true);
+    }
+
+
+    @Test
+    public void resetTest()
```
Comment (bkita) [inline src/test/java/Traffic/ParametrizedRezerwationTest.java:85]: test name should be more descriptive

File: src/test/java/Traffic/ReservationAvaliableParametrizedTest.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+        parking = new Parking(10);
+        reservation = new Reservation(parking);
+        reservation.reset();
+    }
+
+    @After
+    public void clear() {
+        reservation.reset();
+    }
+
```
Comment (bkita) [inline src/test/java/Traffic/ReservationAvaliableParametrizedTest.java:47]: delete empty line

File: src/test/java/Traffic/ReservationAvaliableParametrizedTest.java [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+        this.currentMinTo1 = minTo1;
+        this.currentHourFrom2 = hourFrom2;
+        this.currentMinFrom2 = minFrom2;
+        this.currentHourTo2 = hourTo2;
+        this.currentMinTo2 = minTo2;
+        this.currentStatus = currentStatus;
+
+    }
+
+
```
Comment (bkita) [inline src/test/java/Traffic/ReservationAvaliableParametrizedTest.java]: empty lines?

File: src/test/java/Traffic/ReservationAvaliableParametrizedTest.java [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    public void differentSpaceReservationSameTime()
+    {
+        //Given Empty Parking
+        //When
+        reservation.reservSpace(1, new Time(currentHourFrom1,currentMinFrom1,currentHourTo1,currentMinTo1));
+        //Then
+        Assert.assertEquals(reservation.avaliable(2,new Time(currentHourFrom2,currentMinFrom2,currentHourTo2,currentMinTo2)),true);
+    }
+
+
```
Comment (bkita) [inline src/test/java/Traffic/ReservationAvaliableParametrizedTest.java]: empty lines

File: src/test/java/Traffic/ReservationAvaliableParametrizedTest.java [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    public static Collection inputData()
+    {
+        return Arrays.asList(new Object[][]
+                {{1  ,10	,12	,15	    ,12	,15,    13,	45,	true},
+                {15  ,23	,16	,57	    ,11	,51,    17,	30,	false},
+                {9   ,30	,10	,15	    ,10	,0,	    10,	15,	false},
+                {20  ,15	,21	,30	    ,19	,54,    21,	0,	false}});
+
+    }
+
```
Comment (bkita) [inline src/test/java/Traffic/ReservationAvaliableParametrizedTest.java]: empty lines

File: src/test/java/Traffic/ReservationTest.java [hunk 13]
```diff
@@ -197,6 +197,58 @@ public void timeConflict()
...
+    }
+
+    @Test
+    public void avaliableNotEmptyReservedInImpossibleReserveTime()
+    {
+        reservation.reservSpace(5,new Time(15,30,17,16));
+
+        Assert.assertEquals(reservation.avaliable(5, new Time (17,15,18,40)), false);
+    }
+
```
Comment (bkita) [inline src/test/java/Traffic/ReservationTest.java]: empty lines

File: src/test/java/Traffic/TimeTest.java [hunk 14]
```diff
@@ -1,16 +1,71 @@
...
+                return Arrays.asList(new Object[][]
+        {{14,    20, 15,	10, true},
+        { 0,    0,  0,  59, true},
+        { 12,   0, 12, 0,  true},
+        { 15,   2,  17, 40, true},
+        { 0,    0,  23, 59, true}});
     }
 
 
+
```
Comment (bkita) [inline src/test/java/Traffic/TimeTest.java]: empty lines

**Review by bkita — CHANGES_REQUESTED** [review]

**Comment by salci1990** [comment]
https://github.com/salci1990/RESTassuredTests

**Comment by salci1990** [comment]
strefakursow.pl
bartosz.kita@gmail.com
0MyPython0123!@#

**Pull request merged**